pkgs/applications/editors/eclipse/default.nix

```diff
@@ -227,6 +227,42 @@ rec {
     };
   };
   eclipse_sdk_37 = eclipse-sdk-37; # backward compatibility, added 2016-01-30
+  
+  ### Eclipse Java
+  
+  eclipse-java-47 = buildEclipse {
```

Comment:
Since this is a new package and there is no concern about accommodating people preferring to stay on an older version I think we can remove the 4.7 version here.

Reply:
Ok